src/java.cpp

```diff
@@ -14,7 +14,8 @@
 
 #define DYNAMIC_PROXY_JS_ERROR -4
 
-long v8ThreadId;
+long v8ThreadIdWIN32;
```

Comment:
I went ahead and took the logic out of the condition and put it into a function v8ThreadIdEquals instead. Everything is working perfectly for me on z/OS and OSX